src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProjectProvider.cs

```diff
@@ -63,8 +63,6 @@ public sealed class LegacyPackageReferenceProjectProvider : INuGetProjectProvide
         {
             Assumes.Present(vsProjectAdapter);
 
-            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Only async methods called here need the UI thread, this method itself doesn't have a UI thread affinity.